faces/bin/run-tck.sh

```diff
@@ -0,0 +1,376 @@
+    case "${opt}" in
+        c)
+            clean=true
+            ;;
+        C)
+            clean=true
+            cleanOnly=true
+            ;;
+        v)
+            UNZIP_ARGS="-o"
+            MVN_ARGS=""
+            verboseArgs="-v"
+            ;;
+        \?)
+            echo "Invalid option: -${OPTARG}" >&2
+            printHelp
+            exit 1
+            ;;
+        :)
+            echo "Option -${OPTARG} requires an argument" >&2
+            exit 1
+            ;;
+    esac
+done
+
+# Configure the working directory
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+BASE_DIR="$(dirname "${SCRIPT_DIR}")"
+WORK_DIR="${BASE_DIR}/work"
+# Clean if applicable
+if [ ${clean} == true ]; then
+    rm -rf ${verboseArgs} "${WORK_DIR}"
+    if [ ${cleanOnly} == true ]; then
+        exit
+    fi
+fi
+# Create the directory if required
+if [ ! -d "${WORK_DIR}" ]; then
+    echo "Creating work directory ${WORK_DIR}"
+    mkdir -p "${WORK_DIR}"
+fi
+if [ -z "${TCK_VERSION}" ]; then
+    TCK_VERSION="4.0.1"
+fi
+TCK_ZIP="${WORK_DIR}/jakarta-faces-tck-${TCK_VERSION}.zip"
+TCK_URL=https://download.eclipse.org/jakartaee/faces/4.0/jakarta-faces-tck-${TCK_VERSION}.zip
+TCK_HOME="${WORK_DIR}/faces-tck-${TCK_VERSION}"
+TCK_ROOT=$TCK_HOME/tck
+WILDFLY_HOME=wildfly/target/wildfly
+NEW_WILDFLY="${WORK_DIR}/servers/new-wildfly"
+OLD_WILDFLY="${WORK_DIR}/servers/old-wildfly"
+
+################################################
+# Install WildFly if not previously installed. #
+################################################
+
+# TODO - Override WildFly Version
+
+if [[ -n $JBOSS_HOME ]] 
+then
+    if test -d $JBOSS_HOME 
+    then
+        echo "Using existing server installation " $JBOSS_HOME
+        WILDFLY_HOME=$JBOSS_HOME
+    else
+        echo "JBOSS_HOME points to invalid location " $JBOSS_HOME
+        exit 1
+    fi
+else
+    echo "JBOSS_HOME Is NOT Set."
+    if ! test -d $WILDFLY_HOME
+    then
+        WILDFLY_HOME="${WORK_DIR}/wildfly"
+        echo "Provisioning WildFly to ${WILDFLY_HOME}."
+        pushd "${BASE_DIR}/wildfly"
+        # We still the configuration here as we there is nothing known at this time we can configure
+        mvn ${MVN_ARGS} install -Dprovision.skip=false -Dconfigure.skip=true -Dwildfly.home="${WILDFLY_HOME}"
+        popd
+    fi
+fi
+# At this point WILDFLY_HOME points to the clean server.
+
+####################################
+# Create a copy to run the new TCK #
+####################################
+
+# First delete any existing clone.
+if test -d "${WORK_DIR}"/servers
+then
+    echo "Deleting existing 'servers' directory."
+    rm -fR ${verboseArgs} "${WORK_DIR}"/servers
+fi
+
+mkdir ${verboseArgs} "${WORK_DIR}"/servers
+echo "Cloning WildFly  $WILDFLY_HOME $NEW_WILDFLY"
+cp -R ${verboseArgs} $WILDFLY_HOME $NEW_WILDFLY
+
+echo "skip provisioning of $NEW_WILDFLY (just use defaults and later delete wildfly/pom.xml + wildfly/configure-server.cli."
```

Comment:
This section is just about configuring the existing server not provisioning the server, deleting the pom would prevent the ability to provision above.

In testing the security TCKs I found the provisioning made it very quick to recreate the server being copied for testing after I had worked on a single module - I didn't need to run a full WildFly build to create a distribution.